addons/product/models/product_template.py

```diff
@@ -145,6 +145,11 @@ def _compute_currency_id(self):
         for template in self:
             template.currency_id = template.company_id.currency_id.id or main_company.currency_id.id
 
+    @api.constrains('categ_id')
+    def _check_category(self):
+        if self.categ_id.type == 'view':
+            raise ValidationError(_("You can not create product with 'view' type internal category."))
```

Comment:
Please change the string into : ""You cannot create a product if with the "view" type internal category""